### PR TITLE
Add assign groups in DG

### LINKED ISSUE #48: V1.3 Developer Guide
### Details
- Update the developer guide with the relevant changes
  - Update UML diagrams and sequence diagram with the code changes made
  - Update use cases if necessary

### PR DESCRIPTION
**Pull request by Kurtyjlee**
- Added assign groups to events into the DG
- Fix some diagram images

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -230,6 +218,37 @@ for empty/null inputs in the Person object by checking if the optional field is
...
+
+Each event stores a list of persons assigned to it. The person(s) are represented by their `Name` stored in FumbleLog. This is because the `Name` is the only unique identifier for a person.
+
+When a person is assigned to an event, the person's `Name` is added to the event's list of assigned persons. When a person is unassigned from an event, the person's `Name` is removed from the event's list of assigned persons. When a person's `Name` is modified, the change is also reflected in the event(s) that they are previously assigned to.
+
+Users can assign multiple names to an event by using multiple `n/` identifiers following with the `Name` specified. The `ModelManager` will perform checks on whether the names supplied are valid, i.e the `Name` currently exists in FumbleLog.
+
+When editing the event, specifying `n/` with a `Name` will append this new name to the current list rather than replace the previous names. This is to facilitate the user to assign more persons without accidentally deleting the previous persons assigned. To un-assign a person, the user must manually specify `u/` with the `Name` to un-assign the person from the event.
```
Comment (nicleongyj) [inline docs/DeveloperGuide.md:231]: Nice detailed explanation on the implementation!

File: docs/images/AddPersonSequenceDiagram.png
Comment (nicleongyj) [inline docs/images/AddPersonSequenceDiagram.png:1]: Nice sequence diagram! Can consider splitting it up into smaller diagrams for readability.

**Review by nicleongyj — APPROVED** [review]
LGTM!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -198,7 +186,7 @@ Except for the name, all the fields given to the `add` command are optional.
 
 The flow for the `add` command is described by the following sequence diagram:
 
-<puml src="diagrams/"AddPersonSequenceDiagram.puml" alt="AddPersonSequenceDiagram" />
+<img src="images/AddPersonSequenceDiagram.png" alt="AddPersonSequenceDiagram" width=600 />
```
Comment (leezhengjing) [inline docs/DeveloperGuide.md:189]: Thanks for fixing the missing diagram!

**Review by leezhengjing — APPROVED** [review]
LGTM!

**Pull request merged**